src/components/molecules/Visualizer/Engine/Cesium/core/Indicator.tsx

```diff
@@ -0,0 +1,133 @@
+      let position: Cartesian3 | undefined = undefined;
+      const boundingSphere = new BoundingSphere();
+      const state = (viewer.dataSourceDisplay as any).getBoundingSphere(
+        selected,
+        true,
+        boundingSphere,
+      );
+      // https://github.com/CesiumGS/cesium/blob/main/Source/DataSources/BoundingSphereState.js#L24
+      if (state !== 2 /* BoundingSphereState.FAILED */) {
+        position = boundingSphere.center;
+      } else if (selected.position) {
+        position = selected.position.getValue(viewer.clock.currentTime, position);
+      }
+
+      if (position) {
+        const pos = SceneTransforms.wgs84ToWindowCoordinates(viewer.scene, position);
+        setPos(pos);
+        setIsVisible(true);
+      } else {
+        setIsVisible(false);
+      }
+    };
+
+    viewer.clock.onTick.addEventListener(handleTick);
+    return () => {
+      if (viewer.isDestroyed()) return;
+      viewer.clock.onTick.removeEventListener(handleTick);
+    };
+  }, [viewer]);
+  // console.log( indicator_type );
+  return transiton !== "unmounted" && pos ? (
+    indicator_type === indicatorTypes.crosshair ? (
+      <StyledIcon
+        icon="crosshair"
+        className={className}
+        transition={transiton}
+        style={{ left: pos.x + "px", top: pos.y + "px" }}
+      />
+    ) : indicator_type === indicatorTypes.default ? (
+      <I
+        className={className}
+        transition={transiton}
+        style={{ left: pos.x + "px", top: pos.y + "px" }}
+      />
+    ) : (
+      <Image
+        src={img}
+        width={w}
+        height={h}
+        transition={transiton}
+        style={{ left: pos.x + "px", top: pos.y + "px" }}
+      />
+    )
+  ) : null;
+}
+
+const I = styled.div<{ transition: TransitionStatus }>`
+  position: absolute;
+  width: 50px;
+  height: 50px;
+  transform: translate(-50%, -50%);
+  transition: ${({ transition }) =>
+    transition === "entering" || transition === "exiting" ? "all 0.5s ease" : ""};
+  opacity: ${({ transition }) => (transition === "entering" || transition === "entered" ? 1 : 0)};
+`;
+const StyledIcon = styled(Icon)<{ transition: TransitionStatus }>`
+  position: absolute;
+  width: 50px;
+  height: 50px;
+  transform: translate(-50%, -50%);
+  transition: ${({ transition }) =>
+    transition === "entering" || transition === "exiting" ? "all 0.5s ease" : ""};
+  opacity: ${({ transition }) => (transition === "entering" || transition === "entered" ? 1 : 0)};
+`;
+const Image = styled.img<{ transition: TransitionStatus }>`
+  max-width: 100%;
+  position: absolute;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  background-color: rgba(0, 0, 0, 0.5);
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  transition: ${({ transition }) =>
+    transition === "entering" || transition === "exiting" ? "all 0.5s ease" : ""};
+  opacity: ${({ transition }) => (transition === "entering" || transition === "entered" ? 1 : 0)};
```

Comment:
Is it correct that the animation should only fade in and out? What about animations that change scale? @lavalse

Reply:
@lavalse  would please check

Reply:
@rot1024  , please note that red add his comment and we fixed pr accordingly so I will resolve this one , is that ok ?